src/hypervisor_bootloader/src/dtb.rs

```diff
@@ -602,3 +602,111 @@ impl DtbAnalyser {
+    new_dtb_header.size_dt_struct = original_dtb_header.size_dt_struct;
+    new_dtb_header.size_dt_strings = original_dtb_header.size_dt_strings;
+
+    // copy reserved address section and add new reserved section
+    let original_reserved_address_section =
+        original_base_address + u32::from_be(original_dtb_header.off_mem_rsv_map) as usize;
+    let new_reserved_address_section = new_base_address + total_new_size;
+    let mut pointer = original_reserved_address_section;
+    loop {
+        let address = unsafe { *(pointer as *const u64) };
+        pointer += core::mem::size_of::<u64>();
+        let size = unsafe { *(pointer as *const u64) };
+        pointer += core::mem::size_of::<u64>();
+        if address == 0 && size == 0 {
+            break;
+        }
+    }
+    let reserved_section_size =
+        pointer - original_reserved_address_section - core::mem::size_of::<u64>() * 2;
+    total_new_size += reserved_section_size + core::mem::size_of::<u64>() * 4;
+    if new_size < total_new_size {
+        return Err(());
+    }
+    unsafe {
+        core::ptr::copy_nonoverlapping(
+            original_reserved_address_section as *const u8,
+            new_reserved_address_section as *mut u8,
+            reserved_section_size,
+        );
+    }
+    unsafe {
+        let new_rs_address_area = new_reserved_address_section + reserved_section_size;
+        let new_rs_size_area =
+            new_reserved_address_section + reserved_section_size + core::mem::size_of::<u64>();
+        *(new_rs_address_area as *mut usize) = reserved_address.to_be();
+        *(new_rs_size_area as *mut usize) = reserved_size.to_be();
+        let new_rs_end_area =
+            new_reserved_address_section + reserved_section_size + core::mem::size_of::<u64>() * 2;
+        *(new_rs_end_area as *mut usize) = 0;
+        *((new_rs_end_area + core::mem::size_of::<u64>()) as *mut usize) = 0;
+    }
+
+    // copy struct section
+    let new_struct_base_address = new_base_address + total_new_size;
+    let original_struct_base_address =
+        original_base_address + u32::from_be(original_dtb_header.off_dt_struct) as usize;
+    let struct_section_size = u32::from_be(original_dtb_header.size_dt_struct) as usize;
+    total_new_size += struct_section_size;
+    if total_new_size > new_size {
+        return Err(());
+    }
+    unsafe {
+        core::ptr::copy_nonoverlapping(
+            original_struct_base_address as *const u8,
+            new_struct_base_address as *mut u8,
+            struct_section_size,
+        );
+    }
+
+    // copy string section
+    let new_string_section_address = new_base_address + total_new_size;
+    let original_string_section_address =
+        original_base_address + u32::from_be(original_dtb_header.off_dt_strings) as usize;
+    let string_section_size = u32::from_be(original_dtb_header.size_dt_strings) as usize;
+    total_new_size += (string_section_size + 3) / 4 * 4;
```

Comment:
Could you add a comment to explain the magic numbers? (Sorry I missed this in review comment.)